src/V3EmitMkJson.cpp

```diff
@@ -0,0 +1,354 @@
+            ident += tab;
+            scope.push(type == '{' ? '}' : ']');
+            empty = true;
+            return *this;
+        }
+
+        Printer& put(const std::string& name, const std::string& value) {
+            if (!empty) {
+                *of << ",\n";
+            }
+            *of << ident << "\"" << name << "\": \"" << value << "\"";
+            empty = false;
+            return *this;
+        }
+
+        Printer& put(const std::string& name, bool value) {
+            if (!empty) {
+                *of << ",\n";
+            }
+            *of << ident << "\"" << name << "\": \"" << (value ? "true" : "false") << "\"";
+            empty = false;
+            return *this;
+        }
+
+        Printer& put(const std::string& name, int value) {
+            if (!empty) {
+                *of << ",\n";
+            }
+            *of << ident << "\"" << name << "\": \"" << value << "\"";
+            empty = false;
+            return *this;
+        }
+
+        Printer& begin(char type = '{') {
+            if (!empty) {
+                *of << ",\n";
+            }
+            *of << ident << type << "\n";
+            ident += tab;
+            scope.push(type == '{' ? '}' : ']');
+            empty = true;
+            return *this;
+        }
+
+        Printer& put(const std::string& value) {
+            if (!empty) {
+                *of << ",\n";
+            }
+            *of << ident << "\"" << value << "\"";
+            empty = false;
+            return *this;
+        }
+
+        Printer& put(bool value) {
+            if (!empty) {
+                *of << ",\n";
+            }
+            *of << ident << (value ? "true" : "false");
+            empty = false;
+            return *this;
+        }
+
+        Printer& put(int value) {
+            if (!empty) {
+                *of << ",\n";
+            }
+            *of << ident << value;
+            empty = false;
+            return *this;
+        }
+
+        template <typename T>
+        Printer& putList(const std::string& name, const T& list) {
+            if (list.empty()) {
+                return *this;
+            }
+
+            begin(name, '[');
+
+            for (auto it = list.begin(); it != list.end(); ++it) {
+                put(*it);
+            }
+
+            return end();
+        }
+
+        Printer& end() {
+            assert(ident.length() >= tab.length());
+            ident.erase(ident.end() - tab.length(), ident.end());
+            assert(!scope.empty());
+            *of << "\n" << ident << scope.top();
+            scope.pop();
+            return *this;
+        }
+
+        Printer& operator += (Printer& cursor) {
+            // meaningless syntax sugar, at least for now
+            return *this;
+        }
+
+        const std::unique_ptr<std::ofstream>& of;
+        std::stack<char> scope;
+        std::string ident;
+        std::string tab;
+        bool empty = true;
```

Comment:
Move to top of this class, add `m_` prefix, and comment each.

```suggestion
        // MEMBERS
        const std::unique_ptr<std::ofstream>& of;
        std::stack<char> scope;
        std::string ident;
        std::string tab;
        bool empty = true;
```